### ISSUE TITLE
Mapping of `val_dice_loss` to the loss function `bce_dice_loss`

### ISSUE THREAD

**Issue by mgooty** [petrosgk/Kaggle-Carvana-Image-Masking-Challenge#10]
While defining various callbacks, the `monitor` property is set to a string value `val_dice_loss`. And, while defining the model, the `loss` property is set to `bce_dice_loss` function. But, how does Keras come to know that the `monitor` value `val_dice_loss` corresponds to `bce_dice_loss` function?

**Comment by petrosgk** [comment]
Thanks for spotting this. Keras callbacks always calculated and monitored dice_loss for validation instead of bce_dice_loss or whatever loss the model was defined with. It's now fixed to correctly monitor the defined loss.
